sys/include/net/gnrc/netreg.h

```diff
@@ -212,6 +212,45 @@ typedef struct gnrc_netreg_entry {
     } target;                   /**< Target for the registry entry */
 } gnrc_netreg_entry_t;
 
+/**
+ * @brief   The global locking of netregs
+ *
+ * A shared lock must be held across calls to @ref gnrc_netreg_lookup, and
+ * references obtained through that may only be used for the duration for which
+ * the lock is held.
+ *
+ * The shared lock is counting (i.e. multiple sections of code can acquire it
+ * concurrently), and may be held across blocking operations, as long as they
+ * don't involve the exclusive lock (e.g. by blocking on a socket creation).
+ * It's generally good practice to release them as soon as possible.
+ *
+ * There is an exclusive counterpart to the lock, which is
+ * internal to netreg (and used through functions such as @ref
+ * gnrc_netreg_register and @ref gnrc_netreg_unregister). The current
+ * implementation priorizes shared locks (unlike the rwlock @ref
+ * pthread_rwlock_t, which is not used here). This means that shared locks are
```

Comment:
If it's not used, why reference it here? In my opinion this only leads to confusion.

Reply:
Because it is effectively an RW-lock, and if I came to this as a RIOT user who's seen that we do have `pthread_rwlock_t`, I'd expect it to behave like this, and it won't.

Reply:
Given that your reply confuses me more than it is convincing me, I have the suspicion, I might be right :stuck_out_tongue:

Reply:
But not insisting on this being removed.

Reply:
Not insisting either :-)

But I see your point -- removed now.